x/wasm/keeper/msg_dispatcher.go

```diff
@@ -209,18 +209,19 @@ func redactError(err error) error {
 		return err
 	}
 
+	// If it is a DeterministicError, we can safely return it without redaction.
+	// We only check the top level error to avoid changes in the error chain becoming
+	// consensus-breaking.
+	if _, ok := err.(types.DeterministicError); ok {
+		return err
```

Comment:
Should we unwrap here?

Reply:
We could, but it doesn't make much of a difference since the DeterministicError has the same `Error()` message as its wrapped error.

Reply:
I see, thanks